src/uncrustify.cpp

```diff
@@ -300,7 +298,12 @@ static void version_exit(void)
 }
 
 
-static void redir_stdout(const char *output_file)
+// VS throws an error if an attribute newer than the requested standard level
+// is used; everyone else just ignores it (or warns) like they are supposed to
+#if __cplusplus >= 201703L || !defined(_MSC_VER)
```

Comment:
Note: I *could* just make this `#if __cplusplus >= 201703L`, but versions of GCC and clang new enough to recognize the attribute act on it even at `-std=c++11` (which, IMO, is good), and I would prefer not to over-complicate the PP conditional here.

OTOH, if the `-Wattribute` warning (only from compilers too old to know about `[[nodiscard]]`) is going to bother you, I can change it...